### PR TITLE
v1: Initial implementation

### LINKED ISSUE #2: Indexer Rust implementation

### LINKED ISSUE #8: Support parameter to start retrieving blobs from given block/slot number
Right now the indexer tries to index all blobs from slot 0 which takes ~30min to sync in devnet-v4.

### PR DESCRIPTION
**Pull request by PJColombo**
Resolves #2 , resolves #8 
Notes:

- Last slot tracking database updates are performed after an error occurs or iterating over a set of slots (default to processing all the slots from 0 to the current one) instead of updating it after each slot individually to reduce slot processing overhead.

### CODE REVIEW

**Review by Evalir — COMMENTED** [review]
Overall it's coming together! A few comments, mainly about being a tad more idiomatic about some things. Feel free to close/ignore what you feel like deferring for later.

File: src/beacon_chain/mod.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+impl BeaconChainAPI {
+    pub fn new(rpc_url: String) -> Self {
+        Self { rpc_url }
+    }
+
+    pub async fn get_block(
+        &self,
+        slot: Option<u32>,
+    ) -> Result<Option<Block>, Box<dyn error::Error>> {
```
[the remark this answers is not in the document]
Comment (PJColombo) [reply src/beacon_chain/mod.rs:21]: I was having a bit of trouble finding the best approach for error handling.

This definitely seems the way to go. Thank you for sharing this crate.

File: src/slots/mod.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+use crate::{
+    context::Context,
+    db::{blob_db_manager::DBManager, mongodb::MongoDBManagerOptions, types::Blob},
+    utils::web3::{calculate_versioned_hash, get_eip_4844_tx, get_tx_versioned_hashes},
+};
+
+type StdErr = Box<dyn error::Error>;
+
+pub async fn process_slots(start_slot: u32, end_slot: u32, context: &mut Context) {
```
[the remark this answers is not in the document]
Comment (PJColombo) [reply src/slots/mod.rs:15]: Agree!

**Comment by PJColombo** [comment]
I created some issues to be solved in future PRs following some of the remaining comments discussed above: #9 , #3, #11. 

Merging this now!

**Pull request merged**